shell/src/test/java/alluxio/cli/fsadmin/command/CapacityCommandTest.java

```diff
@@ -56,25 +56,29 @@ public void capacity() throws IOException {
       List<String> expectedOutput = Arrays.asList("Capacity information for all workers: ",
```

Comment:
having the hardcoded string in test to expect is hard to maintain in the future (e.g., when we change the output format, this needs to be updated totally)
Possible to have util functions help doing so?

Reply:
I think for a long time and found that maybe it is useless to add CapacityCommand unit test and SummaryCommand unit test. I could only do some explicitly checking like before and the tests just like doing nothing. File system commands normally do not have unit tests maybe because of the same reason.

@apc999 @aaudiber Could I not include any unit tests for Report commands except some common methods put in FileSystemAdminShellUtils ?

Reply:
For a command like this, I think it is completely appropriate to have a hardcoded string in the test. The formatting is an important part of the functionality - if the formatting gets messed up, that is a serious bug. Like any functionality, if we change the functionality we need to change the test that tests it. If the burden is high for maintaining the test, I think we just need to write it in a better way. For example, we could put the expected output in a test resource file instead of Java. Then, when we change the formatting, we just need to run the test and update the expected output file. Reviewers can see how the new result looks, and make sure it is what we want. Otherwise, it's hard to tell from reading the code what the formatted output will look like exactly.

In frontend testing (shell is a type of frontend), it's a best practice to do regression testing where screenshots are taken from before and after a code change, and compared to make sure no pixels are changed.

Other shell commands generally have just one or two lines of output, so there isn't much formatting to validate. The report commands are different since they have a lot more formatting, so I see it as much more important to test that they are formatted correctly.

Reply:
Really make sense!!

At least when after I change the code, the hardcoded tests push me to run the command code based on the generated input info list which can help me check if I make output look great.

Report commands so far do not have a really long output, so the tests are not that hard to maintain. It is also good to put the expected output in a resource file.

I add back the test and add shortCapacity() test to test the worker infos that only have one tier (do not want total column in that circumstances).